src/fusion_trees/fusiontree.jl

```diff
@@ -0,0 +1,326 @@
+  x = findfirst(==(s3), blocklabels(full_space))
+  isnothing(x) && return 0  # OR labelled(0, s3)?
+  return Int(blocklengths(full_space)[x])
+end
+
+function outer_multiplicity_split(
+  sec1::S, sec2::S, fused::S, outer_mult_index::Integer
+) where {S<:SectorProduct}
+  args1 = arguments(sec1)
+  args2 = arguments(sec2)
+  args12 = arguments(fused)
+  nsymbols = Tuple(map(nsymbol, args1, args2, args12))  # CartesianIndices requires explicit Tuple
+  return Tuple(CartesianIndices(nsymbols)[outer_mult_index])
+end
+
+# --------------- Build trees  ---------------
+# zero leg: need S to get sector type information
+function SectorFusionTree{S}() where {S<:AbstractSector}
+  return SectorFusionTree((), (), trivial(S), (), ())
+end
+function SectorFusionTree{S}(::Tuple{}, ::Tuple{}) where {S<:AbstractSector}
+  return SectorFusionTree((), (), trivial(S), (), ())
+end
+
+# one leg
+function SectorFusionTree(sect::AbstractSector, arrow::Bool)
+  return SectorFusionTree((sect,), (arrow,), sect, (), ())
+end
+
+function braid_tuples(t1::Tuple{Vararg{Any,N}}, t2::Tuple{Vararg{Any,N}}) where {N}
+  t12 = (t1, t2)
+  nested = ntuple(i -> getindex.(t12, i), N)
+  return flatten_tuples(nested)
+end
+
+function grow_tree(
+  parent_tree::SectorFusionTree,
+  branch_sector::AbstractSector,
+  level_arrow::Bool,
+  child_root_sector,
+  outer_mult,
+)
+  child_leaves = (leaves(parent_tree)..., branch_sector)
+  child_arrows = (arrows(parent_tree)..., level_arrow)
+  child_branch_sectors = (branch_sectors(parent_tree)..., root_sector(parent_tree))
+  child_outer_mul = (outer_multiplicity_indices(parent_tree)..., outer_mult)
+  return SectorFusionTree(
+    child_leaves, child_arrows, child_root_sector, child_branch_sectors, child_outer_mul
+  )
+end
+
+function grow_tree(
+  parent_tree::SectorFusionTree, branch_sector::AbstractSector, level_arrow::Bool
+)
+  new_space = fusion_product(root_sector(parent_tree), branch_sector)
+  return mapreduce(vcat, zip(blocklabels(new_space), blocklengths(new_space))) do (la, n)
+    return [
+      grow_tree(parent_tree, branch_sector, level_arrow, la, outer_mult) for
+      outer_mult in 1:n
+    ]
+  end
+end
+
+function build_trees(old_trees::Vector, sectors_to_fuse::Tuple, arrows_to_fuse::Tuple)
+  next_level_trees = mapreduce(vcat, old_trees) do tree
+    return grow_tree(tree, first(sectors_to_fuse), first(arrows_to_fuse))
+  end
+  return build_trees(
+    next_level_trees, Base.tail(sectors_to_fuse), Base.tail(arrows_to_fuse)
+  )
+end
+
+function build_trees(trees::Vector, ::Tuple{}, ::Tuple{})
+  return trees
+end
+
+function build_trees(
+  sectors_to_fuse::NTuple{N,<:AbstractSector}, arrows_to_fuse::NTuple{N,Bool}
+) where {N}
+  trees = [SectorFusionTree(first(sectors_to_fuse), first(arrows_to_fuse))]
+  return build_trees(trees, Base.tail(sectors_to_fuse), Base.tail(arrows_to_fuse))
+end
+
+# --------------- convert to Array  ---------------
+to_tensor(::SectorFusionTree{<:Any,0}) = ones(1)
```

Comment:
I would call it `to_array` just to stick to Julia notation, I assume it is always outputting some kind of `AbstractArray`.

Reply:
Both of those make sense, we could define both, i.e.:
```julia
Base.Array(t::SectorFusionTree) = ...
Base.AbstractArray(t::SectorFusionTree) = Array(t)
```